src/InitiateState.cpp

```diff
@@ -400,7 +415,7 @@ void* __fastcall do_game_update_new(void* thislol, int edx, int* a, int* b){
 		return do_game_update(thislol, a, b);
 	}
 
-	lua_State* L = (lua_State*)*((void**)thislol);
+	lua_State* L = reinterpret_cast<lua_State *>(thislol);
```

Comment:
Does this really work? I was specifically casting to a pointer-pointer, then derefencing to a pointer, not just casting to pointer. I was derefencing the 'this' pointer to get the first variable contained therein, I don't think the lua state is in the same spot?